### PR TITLE
Added handler to complete scan tasks when terminated unexpectedly

### LINKED ISSUE #431: [Bug] The batch scan task will not have scan DB state updated when terminated by Batch service
The Batch service will terminate long running scan tasks. In this case no corresponding state will be updated within Cosmom DB. The scan state will be marked as 'running'.

There should be a mechanism to handle batch scan tasks state after external termination. Either job manager should re-validate tasks states periodically or job scheduler post task should do re-validation.

### PR DESCRIPTION
**Pull request by lamaks**
#### Description of changes

Preserve task arguments object as part of task environment variable
Complete terminated tasks by setting correct scan state in Cosmos DB
Delete obsolete install script
Fix Mac OS bash compatibility issue

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #431, #491
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/web-api-scan-job-manager/src/worker/worker.ts [hunk 1]
```diff
@@ -76,18 +81,58 @@ export class Worker {
...
+                    let pageScanResult = await this.onDemandPageScanRunResultProvider.readScanRun(taskArguments.id);
+                    if (pageScanResult !== undefined) {
+                        if (pageScanResult.run.state !== 'failed') {
+                            pageScanResult.run = {
+                                state: 'failed',
+                                timestamp: failedTask.timestamp.toJSON(),
+                                error,
+                            };
+                            pageScanResult = await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+                        }
```
Comment (devaradhanm) [inline packages/web-api-scan-job-manager/src/worker/worker.ts]: can you log here the failure info from batch here? incase we have 2 different failures (scan failure message is different from task failure)

Comment (lamaks) [reply packages/web-api-scan-job-manager/src/worker/worker.ts]: batch error logged below
`                    this.logger.logError(error, { taskProperties: JSON.stringify(failedTask) });
`

File: packages/web-api-scan-job-manager/src/worker/worker.spec.ts [hunk 2]
```diff
@@ -39,6 +38,7 @@ describe(Worker, () => {
     let serviceConfigMock: IMock<ServiceConfiguration>;
     let loggerMock: IMock<MockableLogger>;
     let batchPoolLoadSnapshotProviderMock: IMock<BatchPoolLoadSnapshotProvider>;
+    let onDemandPageScanRunResultProviderMock: IMock<OnDemandPageScanRunResultProvider>;
```
Comment (devaradhanm) [inline packages/web-api-scan-job-manager/src/worker/worker.spec.ts]: i don't see onDemandPageScanRunResultProviderMock.verifyAl() call anywhere. Without this call, the verifiable setups won't be validated.

File: packages/azure-services/src/azure-batch/batch.spec.ts [hunk 3]
```diff
@@ -98,6 +116,171 @@ describe(Batch, () => {
...
+            taskMock
+                .setup(async o => o.list(config.jobId, options))
+                .returns(async () => Promise.resolve(<TaskListResponse>(<unknown>items1)))
+                .verifiable();
+            taskMock
+                .setup(async o => o.listNext(items1.odatanextLink, options))
+                .returns(async () => Promise.resolve(<TaskListResponse>(<unknown>items2)))
+                .verifiable();
+
+            const expectedFailedTasks: BatchTask[] = [
```
Comment (devaradhanm) [inline packages/azure-services/src/azure-batch/batch.spec.ts]: can you refactor this & create a helper method to generate these objects?

File: packages/web-api-scan-job-manager/src/worker/worker.spec.ts (hunk 2 above)
Comment (lamaks) [reply packages/web-api-scan-job-manager/src/worker/worker.spec.ts]: `afterEach(() => {
        batchMock.verifyAll();
        queueMock.verifyAll();
        serviceConfigMock.verifyAll();
        batchPoolLoadSnapshotProviderMock.verifyAll();
        systemMock.verifyAll();
        onDemandPageScanRunResultProviderMock.verifyAll();
    });`

File: packages/azure-services/src/azure-batch/batch.spec.ts (hunk 3 above)
Comment (lamaks) [reply packages/azure-services/src/azure-batch/batch.spec.ts]: No, generalization in UTs is not always the best and will not help future understanding of test data/test flow. In this case it has no high benefits either as only few properties are common in array of objects. The simple declaration is only required here as well.

**Pull request merged**